chapter_preliminaries/ndarray.md

```diff
@@ -659,6 +736,11 @@ Z.assign(X + Y)
 print('id(Z):', id(Z))
 ```
 
+```{.python .input}
+#@tab jax
+# JAX arrays do not allow in-place operations
```

Comment:
We could add a section above about how JAX handles this. Something like:

:begin_tab:`jax`
JAX arrays do not allow in-place operations. Instead, the [`at`](https://jax.readthedocs.io/en/latest/_autosummary/jax.numpy.ndarray.at.html?highlight=in%20place#jax-numpy-ndarray-at) property provides a functionally pure equivalent of in-place array modificatons. Expressions like `x = x.at[idx].set(y)` are guaranteed to be applied in-place inside a `jit()` compiled function.
:end_tab:

Reply:
Hi @ghost, yes that would be ideal in this case. But for now, we'll stick to code changes, and edit the text later.